bundles/org.eclipse.jface.text/src/org/eclipse/jface/text/codemining/SuffixLineContentCodeMining.java

```diff
@@ -0,0 +1,45 @@
+/*******************************************************************************
+ *  Copyright (c) 2024, SAP SE
+ *
+ *  This program and the accompanying materials
+ *  are made available under the terms of the Eclipse Public License 2.0
+ *  which accompanies this distribution, and is available at
+ *  https://www.eclipse.org/legal/epl-2.0/
+ *
+ *  SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.jface.text.codemining;
+
+import java.util.function.Consumer;
+
+import org.eclipse.swt.events.MouseEvent;
+
+import org.eclipse.jface.text.Position;
+
+/**
+ * Line content code mining which is drawn right to the given position. Cursor and selection
+ * handling are not including the code mining.
+ */
+public abstract class SuffixLineContentCodeMining extends LineContentCodeMining {
```

Comment:
I would rather imagine a new flag on LineContentCodeMning than a whole new method here.
IMO, a new constructor `LineContentCodeMining(Position, boolean afterPosition /* default being false in other constructors that don't have this arg */, provider)` would be simpler to both consume and maintain.

Reply:
Thanks a lot for the fast reply. I changed the API as requested and added a further constructor in LineContentCodeMining. I have put "boolean afterPosition" as last contructor argument for compatibility reasons. Do you think it is better to place it as second argument?

Reply:
For compatibility, you only need to keep the current constructors. You're free to design your new ones freely.
So yes, I think it's better to place it as 2nd argument if it's not conflicting with any other existing constructor.

Reply:
ok, I added it as 2nd argument. Is it because first argument is "position" and next argument "afterPosition" is then influencing the position?

Reply:
Yes, the position where the codemining is rendered is a function of those 2 arguments. Grouping them together makes things faster to understand.

Reply:
ok, understood. Makes absolutely sense.